python/cudf/cudf/_libxx/quantiles.pyx

```diff
@@ -15,39 +16,29 @@ from cudf._libxx.types cimport (
 
 from cudf._libxx.cpp.table.table cimport table
 from cudf._libxx.cpp.table.table_view cimport table_view
+from cudf._libxx.cpp.column.column_view cimport column_view
 cimport cudf._libxx.cpp.types as libcudf_types
 cimport cudf._libxx.cpp.quantiles as cpp_quantiles
 
 
 def quantiles(Table source_table,
               vector[double] q,
               object interp,
-              object is_input_sorted,
-              list column_order,
-              list null_precedence):
+              Column sortmap,
```

Comment:
Below, we handle default values for `sortmap` and `cast_to_doubles` in `frame.py` - let's just do that here in the Cython. If we end up calling the `quantiles` Cython in more than one place, we'll end up having to duplicate that logic for handling default values.

Reply:
Unfortunately moving the default `sortmap` to `.pyx` would mean one `pyx` depending on another. It would also mean passing `ascending` and `na_position` to the `pyx` file.

I'm not sure I like that option. Did you see it differently?
```
        sortmap = (
            None
            if is_sorted
            else libcudfxx.sort.order_by(self, ascending, na_position,)
        )
```

Reply:
Also, I'm not sure how we would default `cast_to_doubles` in the `pyx`, since `None` is a valid option.

Reply:
It should be totally fine to pass Python arguments to a `def` function in the `.pyx` file. Anything that needs to be shared between two `.pyx` files can be declared in a `.pxd` file and `cimported` from the `.pyx` files.

Reply:
For example, we don't have to explicitly type `cast_to_doubles` as `bool` here. Leaving it untyped (or alternatively specifying `object` explicitly as its type) will allow us to pass arbitrary Python objects (including `None`).

Reply:
On the one hand, we can get default arguments (for `sortmap`, this would require `kwargs**`, since `None` is a valid option. On the other hand, we can get type checking. I don't know how much mileage we'd get out of default arguments at this level of the API, but I do believe consistent type checking will be beneficial no matter who's invoking the method.

If you're not opposed to it, I'd prefer to keep type checking at the expense of losing default arguments.

Reply:
No problem - I'm not opposed.

Although I will note that we might be less inclined to enforce `bool` types as it precludes the use of non-boolean arguments such as `0`, `1` or `[]`, where it's quite common in Python to use them in place of `True` or `False` in many situations.

Reply:
I was mistaken - it looks like you can still pass arbitrary Python objects for a `bool` typed argument. Python will just call the `__bool__()` method to see if that value is "true" or "false" (defaulting to `True` is `__bool__` is not defined).